examples/vault-iam-auth/user-data-auth-client.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+# This script is meant to be run in the User Data of each EC2 Instance while it's booting. The script uses the
+# run-consul script to configure and start Consul in client mode. Note that this script assumes it's running in an AMI
+# built from the Packer template in examples/vault-consul-ami/vault-consul.json.
+
+set -e
+
+# Send the log output from this script to user-data.log, syslog, and the console
+# From: https://alestic.com/2010/12/ec2-user-data-output/
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+# These variables are passed in via Terraform template interpolation
+/opt/consul/bin/run-consul --client --cluster-tag-key "${consul_cluster_tag_key}" --cluster-tag-value "${consul_cluster_tag_value}"
+
+
+function retry {
+  local readonly cmd=$1
+  local readonly description=$2
+
+  for i in $(seq 1 30); do
+    echo "$description"
+
+    # The boolean operations with the exit status are there to temporarily circumvent the "set -e" at the
+    # beginning of this script which exits the script immediatelly for error status while not losing the exit status code
+    output=$(eval "$cmd") && exit_status=0 || exit_status=$?
+    if [[ $exit_status -eq 0 ]]; then
+      return
+    fi
+    echo "$description failed. Will sleep for 10 seconds and try again."
+    sleep 10
+  done;
+
+  echo "$description failed."
+  exit $exit_status
+}
+
+# Retrying this just in case terraform is still connecting and provisioning these files
```

Comment:
If you package the script via Packer, you can avoid this retry. Moreover, all the dependencies will already be installed, which is more idiomatic to how we build our modules.

Reply:
If it's fine to do it even though it is used just for this example, sure!